src/change_stream.ts

```diff
@@ -923,7 +924,7 @@ export class ChangeStream<
     }
 
     if (isResumableError(changeStreamError, this.cursor.maxWireVersion)) {
-      this.cursor.close();
+      this.cursor.close().catch(() => null);
```

Comment:
Appreciate the suggestion! IIUC wouldn't leaving the promise without a rejection handler risk exiting the process if it did error? it's spec-ed behavior to ignore errors from close (@baileympearson do you have the reference?)

Reply:
I don't know if it spec-ed.  We had a discussion about this on the PR already but I'll post this again with more detail. 

Relevant ticket: [jira.mongodb.org/browse/NODE-4217](https://jira.mongodb.org/browse/NODE-4217)
Related PR: https://github.com/mongodb/node-mongodb-native/pull/3226

I encountered this issue for cursor.close and filed the aforementioned ticket and left the TODOs that you see in the above PR.  When we triaged/refined it, we decided that the current behavior was correct and closed the ticket as won't do (at least, I _thought_ the jira status should be won't do but I guess we kinda ruined that flow by adding AC to the ticket related to removing TODOs.  Maybe that's a process improvement we could make for the future).  You can see Marin's AC on the ticket is to remove the TODOs I added, which I did in the PR.  We don't have any real record of this but I do remember Jeff mentioning that this would be the intended behavior.  My understanding was that the justification was two-fold:

1. What exactly would the user do in the case that cursor.close failed? Try again?  If killCursors fails, there isn't much they can do because it probably means something larger is going wrong.
2. Cursors timeout on the server anyways, so if the cursor fails to close eventually it will be GC'd.

I think we could revisit this though because I think an argument could be made for cursor.close() propagating errors.

-------

Now this particular instance is a separate issue, irrespective whether or not cursor.close() swallows errors.  This call to cursor.close() is inside the error handler for ChangeStreams.  This code is run if there is an error with the change stream.  The desired (but not explicitly spec'd behavior that I know of) would be

1. Error occurs in the CS (resumable or not)
2. We attempt to resume.  This involves cleaning up the previous cursor and restarting the stream
3. Resuming, or failing that, propagating the error to the user

What we don't want is the error from cursor.close() either crashing the user's process (unhandled process rejection) or overridding the original change stream error.  The fact that our CS uses cursors internally and that they're closed during the resume process is an implementation detail and not something that a user cares about.  So in this scenario, irrespective of whether or not cursor.close() swallows errors, we should swallow the error.

Reply:
> IIUC wouldn't leaving the promise without a rejection handler risk exiting the process if it did error?

Right, it does make sense to leave a `.catch(() => {})` in places where you intentionally ignore errors :+1: